### PR TITLE
Deployment.md refreshing

### LINKED ISSUE #789: Deployment.md - Refresh & update
Relates to #769 and https://github.com/btcpayserver/btcpayserver-doc/issues/769#issuecomment-736082738

* Update `deployment.md` file
* While refreshing the way to choose a deployment based in `use-case` and/or `"skill"` of the user.

### PR DESCRIPTION
**Pull request by Zaxounette**
Closes #789 

5e72415
* Deletes old tables
* Creates a new expanded table

b409fba
* Deletes decision making infographic - Cause: Redundancy

b99c470
* Moves "Whats my options ?" paragraphe under newly created table
* Deletes "Whats my options ?" H2 title and replaces it by CTO sentence
* Modifies docker link to point to actual documentation page instead of github page

9db43dc
* Typo
* Deletes "Third-Part Host" unnecessary link (redundant)

### CODE REVIEW

**Comment by Zaxounette** [comment]
`├─BROKEN─ https://mainnet.demo.btcpayserver.org/translate/ (HTTP_404)` is not related top this PR.

That link can be deleted BTW, since the translator is obsolete. Will do another PR for that.

**Comment by britttttk** [comment]
>Deletes decision making infographic

I think we should keep the decision infographic

File: docs/Deployment.md [hunk 1]
```diff
@@ -10,51 +10,30 @@ BTCPay is a non-custodial invoicing system which eliminates the involvement of a
...
-| ---------------------------------------- |:----------------------------:|:----------:|:------------:|
-|Business<br>(Fast Setup)                  |[3rd Party <br> BTCPay Hosts](./ThirdPartyHosting.md)  |[LunaNode <br> Web-Wizard](./LunaNodeWebDeployment.md)* |- Low Difficulty<br>- BTCPayServer Support (1)<br>- Lightning Network (2)     |
-|Cost / Month                                     | Free                    | $10 |BTC Accepted  |
 
-**LunaNode Web-Wizard is a VPS solution, deployable from an easy-web interface.**
+## To chose one that will best suit your needs, consider the following:<br>
 
-<br>
+|Choosing a deployment <br>method      |   Difficulty   |    Production/Dev |      All features <br>available     |             Approx. cost<br> per month      |
+| -------------------------------- |:--------------:|:----------------------:|:---------------:|:-------------:|:-------------:|
```
Comment (britttttk) [inline docs/Deployment.md:19]:
```suggestion
| -------------------------------- |:--------------:|:----------------------:|:---------------:|:-------------:|
```

The github preview of this table doesn't look right. Are you sure you need 6 segments here with only 5 columns?

Comment (Zaxounette) [reply docs/Deployment.md:19]: Oh I must have forgotten to delete that. I had a 6th column before making this PR.

File: docs/Deployment.md [hunk 2]
```diff
@@ -10,51 +10,30 @@ BTCPay is a non-custodial invoicing system which eliminates the involvement of a
...
-|Web Solutions                             |1.                            |2.          |Why?          |
-| ---------------------------------------- |:----------------------------:|:----------:|:------------:|
-|Business<br>(Fast Setup)                  |[3rd Party <br> BTCPay Hosts](./ThirdPartyHosting.md)  |[LunaNode <br> Web-Wizard](./LunaNodeWebDeployment.md)* |- Low Difficulty<br>- BTCPayServer Support (1)<br>- Lightning Network (2)     |
-|Cost / Month                                     | Free                    | $10 |BTC Accepted  |
 
-**LunaNode Web-Wizard is a VPS solution, deployable from an easy-web interface.**
+## To chose one that will best suit your needs, consider the following:<br>
 
-<br>
+|Choosing a deployment <br>method      |   Difficulty   |    Production/Dev |      All features <br>available     |             Approx. cost<br> per month      |
```
Comment (pavlenex) [inline docs/Deployment.md:18]:
```suggestion
|Deployment method <br>method      |   Difficulty   |    Production/Dev |      All features <br>available     |             Approx. cost<br> per month      |
```

**Comment by Zaxounette** [comment]
> I think we should keep the decision infographic

I feel like it's redundant and less readable than the new table or the old ones.
Why would you want it kept ?

**Comment by pavlenex** [comment]
> Why would you want it kept ?

Because Nicolas made it and might kick our ass if we remove it.

File: docs/Deployment.md [hunk 3]
```diff
@@ -10,51 +10,30 @@ BTCPay is a non-custodial invoicing system which eliminates the involvement of a
 Developer deployments are not recommended for production environments and require the user to have technical knowledge related to the build.
```
Comment (pavlenex) [inline docs/Deployment.md:10]: > Developer deployments
It's unclear what developer deployments are?

Comment (Zaxounette) [reply docs/Deployment.md:10]:
```suggestion
Manual deployments and Hardware builds are not recommended for production environments and require the user to have technical knowledge related to the build.
```

This should make it more understandable, and more relating to the titles `Manual deployment` and `Hardware build` shown in the new table.
Thoughts ?

Comment (pavlenex) [reply docs/Deployment.md:10]: Yes, that's clearer indeed. :)

**Comment by Zaxounette** [comment]
> > Why would you want it kept ?
> 
> Because Nicolas made it and might kick our ass if we remove it.

If that's the sole reason, then I call upon the god of all deployments, the might of censorship-resistance, the muse of good-looking-hair, I name @NicolasDorier !

File: docs/Deployment.md (hunk 3 above)
Comment (Zaxounette) [reply docs/Deployment.md:10]: Then committing that :)

**Comment by Zaxounette** [comment]
@britttttk , I f*ed-up my rebase and it introduced commit 262a661 that has nothing to do there. This change is already in master.

**Comment by britttttk** [comment]
@Zaxounette We should keep the infographic because it's much more clear than a table for _how_ to make a decision which is what the document is. 

For example there is no indication as to why a user would choose one deployment method over another for reasons besides price and difficulty. However the infographic clearly displays the decisions a user should consider regarding lightning, business vs learning, etc.

I still nack the idea of production vs. development denotation that is used in the table because it I think it is subjective to the user's capability. However that is up to you.

**Comment by britttttk** [comment]
>I feel like it's redundant and less readable than the new table or the old ones.
Why would you want it kept ?

None of the information shown in the decision graphic appears in the table. Where do you see a redundancy? 

I do find this info to be redundant and duplicate links should be removed and replaced with links in the table:

image

**Comment by Zaxounette** [comment]
> We should keep the infographic because it's much more clear than a table for _how_ to make a decision which is what the document is.
> 
> For example there is no indication as to why a user would choose one deployment method over another for reasons besides price and difficulty. However the infographic clearly displays the decisions a user should consider regarding lightning, business vs learning, etc.

Thanks, I better understand :)
I'll revert the deletion.

> I still nack the idea of production vs. development denotation that is used in the table because it I think it is subjective to the user's capability. However that is up to you.

It is completely subjective I agree. My intent is not so much as to discourage users with the technical skills from going manual, but the users without such skills.
This aims to be a complement to the warnings we put in a short while ago.

> I do find this info to be redundant and duplicate links should be removed and replaced with links in the table

Good point.
Will fix this now.

**Comment by Zaxounette** [comment]
> I still nack the idea of production vs. development denotation that is used in the table because it I think it is subjective to the user's capability. However that is up to you.

In the warnings, we use the terms `technical literacy`. Would this be a more suitable denotation instead of prod./dev. ?
As in `Technical Literacy Required: Yes/No.`

**Comment by britttttk** [comment]
>In the warnings, we use the terms technical literacy. Would this be a more suitable denotation instead of prod./dev. ?
As in Technical Literacy Required: Yes/No.

I see combined with the warning it is clear. I think the infographic makes it increasingly clear with the "Are you a hacker" type questions :D

LGTM

**Review by britttttk — APPROVED** [review]

**Comment by Zaxounette** [comment]
> I see combined with the warning it is clear. I think the infographic makes it increasingly clear with the "Are you a hacker" type questions :D
> 
> LGTM

Nice ! 
Happy you're happy :)

**Pull request merged**